app/modules/parsing/graph_construction/parsing_service.py

```
@@ -63,104 +63,127 @@ async def parse_directory(
+                logger.exception(
+                    f"Error during parsing for project {project_id}",
+                    project_id=project_id,
+                    user_id=user_id,
+                )
+                # Log the formatted traceback string explicitly for detailed debugging
                 logger.error(
-                    f"Failed to update project status after error: {update_error}"
+                    f"Full traceback for project {project_id}:\n{tb_str}",
+                    project_id=project_id,
+                    user_id=user_id,
+                )
```

Ruff (0.14.7)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)